modules/exploits/linux/http/hadoop_unauth_exec.rb

```diff
@@ -0,0 +1,94 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Exploit::CmdStager
+
+  def initialize(info={})
+    super(update_info(info,
+      'Name'           => 'Hadoop YARN ResourceManager Unauthorized Command Execution',
+      'Description'    => %q{
+          This module exploits an unauthorized command execution vulnerability in Apache Hadoop through ResourceManager REST API.
+      },
+      'License'        => MSF_LICENSE,
+      'Author'         =>
+        [
+          'cbmixx',                                 # Proof of concept
+          'Green-m  <greenm.xxoo[at]gmail.com>'     # Metasploit module
+        ],
+      'References'     =>
+        [
+          [ 'URL', 'http://archive.hack.lu/2016/Wavestone%20-%20Hack.lu%202016%20-%20Hadoop%20safari%20-%20Hunting%20for%20vulnerabilities%20-%20v1.0.pdf'],
+          [ 'URL', 'https://github.com/vulhub/vulhub/tree/master/hadoop/unauthorized-yarn']
+        ],
+
+      'Platform'       => 'linux',
+      'Arch'           => [ARCH_X86, ARCH_X64],
+      'Targets'        =>
+        [
+          ['Automatic',  {} ],
+        ],
+      'Privileged'     => false,
+      'DisclosureDate' => 'Oct 19 2016',
+      'DefaultTarget'  => 0
+      ))
+
+    register_options([
```

Comment:
Personal preference. When there's only one module option with a short description, suggest:

```ruby
    register_options([Opt::RPORT(8088)])
# or
    register_options [Opt::RPORT(8088)]
```

Alternatively, as the default option is also an implied option (`RPORT` is implied for `remote` modules); you could bypass `register_options` entirely, and instead use the `DefaultOptions` key :

```ruby
'DefaultOptions' => {'RPORT' => 8088}
```